### ISSUE TITLE
Screen Reader announcing incorrect state for the list items present in Pin column 

### ISSUE THREAD

**Issue by AmyGao123** [ceolter/ag-grid#9582]
**I'm submitting a**
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```
**Providing a Reproducible Scenario**
- Go to https://www.ag-grid.com/example/
- Open Screen Reader (I've used Narrator)
- While navigating with keyboard after expanding pin column option, Screen reader announces as selected for all the items present. for example, "Pin Left" is also announced as selected even when "No Pin" is the selected option.
![image](https://github.com/user-attachments/assets/71b10a61-1fac-4843-8849-1743282af6ad)

**Current behaviour**
Screen reader announces selected for both selected and not selected options.

**Expected behavior**
Screen Reader should announce the correct Selected/Unselected state for the items present in the pin column option.

**Please tell us about your environment:**
Edge, Narrator

**Comment by AG-Zoheil** [comment]
Hi @AmyGao123,

We have been unable to reproduce this. To confirm we followed these steps using NVDA:

1. Open https://www.ag-grid.com/example/
2. Navigate to a column menu and open it
3. Navigate to the pinning menu item
4. Navigate into the sub-menu

**Actual & Expected:** "No Pin checked, 1 of 3, Level 2".

If this is something unique to the screen reader you are using, please get in touch with them to review as it is working with other screen readers.

Kind regards,
Zoheil
